File: apps/ios/App/Parent/Model/ParentModels.swift
import Foundation

enum ParentRole: String {
    case parentGuardian = "parent_guardian"
    case student
    case teacher
}

struct ParentSession: Hashable {
    let userId: String
    let schoolId: String
    let role: ParentRole
    let isDevelopment: Bool
}

struct ParentProfile: Hashable {
    let id: String
    let displayName: String
    let email: String
    let schoolId: String
}

struct SchoolSummary: Hashable {
    let id: String
    let name: String
}

struct ChildSummary: Identifiable, Hashable {
    let studentId: String
    let displayName: String
    let studentNumber: String
    let yearGroup: String
    let className: String
    let schoolId: String

    var id: String { studentId }
}

struct ParentAnnouncement: Identifiable, Hashable {
    let id: String
    let title: String
    let body: String
    let authorDisplayName: String
    let publishedAt: String
    let audienceLabel: String
    let schoolId: String
    let recipientUserId: String
    let relatedStudentIds: Set<String>
    let readAt: String?
}

enum ParentFormStatus: String, Hashable {
    case outstanding
    case submitted
    case closed
}

enum ParentFormQuestionType: String, Hashable {
    case acknowledgement
    case consent
    case shortText
    case longText
    case singleChoice
}

struct ParentFormQuestion: Identifiable, Hashable {
    let id: String
    let label: String
    let type: ParentFormQuestionType
    let required: Bool
    let options: [String]
}

struct ParentFormAnswer: Hashable {
    let questionId: String
    let value: String
}

struct ParentFormTask: Identifiable, Hashable {
    let recipientId: String
    let formId: String
    let title: String
    let description: String
    let deadlineAt: String?
    let schoolId: String
    let recipientUserId: String
    let child: ChildSummary?
    let status: ParentFormStatus
    let questions: [ParentFormQuestion]
    let submittedAt: String?
    let submittedAnswers: [ParentFormAnswer]

    var id: String { recipientId }
}

struct ParentHomeState: Hashable {
    let parent: ParentProfile
    let school: SchoolSummary
    let selectedChild: ChildSummary?
    let children: [ChildSummary]
    let announcements: [ParentAnnouncement]
    let forms: [ParentFormTask]

    var unreadCount: Int {
        announcements.filter { $0.readAt == nil }.count
    }

    var outstandingFormCount: Int {
        forms.filter { $0.status == .outstanding }.count
    }
}

enum LanguagePreference: String, CaseIterable, Identifiable {
    case english = "en"
    case traditionalChinese = "zh-Hant"

    var id: String { rawValue }

    var label: String {
        switch self {
        case .english:
            return "English"
        case .traditionalChinese:
            return "Traditional Chinese"
        }
    }
}

File: apps/android/app/src/main/java/com/example/aischoolplatform/dev/parent/model/ParentModels.kt
package com.example.aischoolplatform.dev.parent.model

enum class ParentRole(val value: String) {
    ParentGuardian("parent_guardian"),
    Student("student"),
    Teacher("teacher")
}

data class ParentSession(
    val userId: String,
    val schoolId: String,
    val role: ParentRole,
    val isDevelopment: Boolean
)

data class ParentProfile(
    val id: String,
    val displayName: String,
    val email: String,
    val schoolId: String
)

data class SchoolSummary(
    val id: String,
    val name: String
)

data class ChildSummary(
    val studentId: String,
    val displayName: String,
    val studentNumber: String,
    val yearGroup: String,
    val className: String,
    val schoolId: String
)

data class ParentAnnouncement(
    val id: String,
    val title: String,
    val body: String,
    val authorDisplayName: String,
    val publishedAt: String,
    val audienceLabel: String,
    val schoolId: String,
    val recipientUserId: String,
    val relatedStudentIds: Set<String>,
    val readAt: String? = null
)

data class ParentHomeState(
    val parent: ParentProfile,
    val school: SchoolSummary,
    val selectedChild: ChildSummary?,
    val children: List<ChildSummary>,
    val announcements: List<ParentAnnouncement>
) {
    val unreadCount: Int = announcements.count { it.readAt == null }
}

enum class LanguagePreference(val code: String, val label: String) {
    English("en", "English"),
    TraditionalChinese("zh-Hant", "Traditional Chinese")
}
